pkg/ingester/metrics.go

```diff
@@ -231,14 +231,12 @@ func newIngesterMetrics(
 			return 0
 		}),
 
-		inflightRequests: promauto.With(r).NewGaugeFunc(prometheus.GaugeOpts{
-			Name: "cortex_ingester_inflight_push_requests",
-			Help: "Current number of inflight push requests in ingester.",
-		}, func() float64 {
-			if inflightRequests != nil {
-				return float64(inflightRequests.Load())
-			}
-			return 0
+		inflightRequests: promauto.With(r).NewSummary(prometheus.SummaryOpts{
+			Name:       "cortex_ingester_inflight_push_requests",
+			Help:       "Number of inflight requests sampled at a regular interval. Quantile buckets keep track of inflight requests over the last 60s.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.75: 0.02, 0.8: 0.02, 0.9: 0.01, 0.95: 0.01, 0.99: 0.001},
+			MaxAge:     time.Minute,
+			AgeBuckets: 6,
 		}),
```

Comment:
I think we should keep both metrics and follow the deprecation process, for the gauge, as users may be slow at upgrading their mixin.

If not, at least call the new summary with a different name, as otherwise a query like `avg (cortex_ingester_inflight_push_requests)` would keep working but providing wrong results.